packages/data-stores/src/site/actions.ts

```diff
@@ -314,7 +314,7 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 				apiNamespace: 'wpcom/v2',
 				body: {
 					trim_content: true,
-					vertical_id: selectedDesign.verticalizable ? siteVerticalId : undefined,
+					vertical_id: selectedDesign.verticalizable && siteVerticalId ? siteVerticalId : undefined,
```

Comment:
**issue**: we still don't have anything that tests (in a deterministic manner) whether this ternary will return the expected results under _all_ conditions. While the E2E tests below are decent additions, it is crucial to remember that E2E tests **are not meant to catch every bug**. I recently responded to another request for more E2E tests [here](pciE2j-1oJ-p2#comment-1145) and similar concepts apply to this PR.

**suggestion**: there should be unit tests written which tests the `setDesignOnSite` method, and more specifically the part where the response is generated and returned:

```
			const response: { blog: string } = yield wpcomRequest( {
				path: `/sites/${ encodeURIComponent( siteSlug ) }/theme-setup`,
				apiNamespace: 'wpcom/v2',
				body: {
					trim_content: true,
					vertical_id: selectedDesign.verticalizable && siteVerticalId ? siteVerticalId : undefined,
					pattern_ids: recipe?.pattern_ids,
					header_pattern_ids: recipe?.header_pattern_ids || [],
					footer_pattern_ids: recipe?.footer_pattern_ids || [],
				},
				method: 'POST',
			} );
```

The important thing here is to ensure that values in the response JSON are being set to the expected & correct values given a range of inputs. 

An example unit test may look something like this (in pseudocode)

```
function test() {
    apiNameSpace === 'wpcom/v2'
    body.vertical_id === expected_value
    body.pattern_ids === expected_value
    body.pattern_ids === header_pattern_ids === []
}
```

Reply:
Thanks 👍 I've added the unit test to the `setDesignOnSite`  action